Add SLIP-0077

stevenroose opened the pull request:
An implementation of https://github.com/satoshilabs/slips/blob/master/slip-0077.md that is compatible with Elements Core and libwally.

RCasatta commented:
gdk electrum will need this, at the moment is done internally with wally bindings.

Do you want to add some more test vectors?
https://github.com/ElementsProject/libwally-core/blob/721b67c061b38bc5ec2c8e5cd41e89357701685d/src/test/test_confidential_addr.py#L12
(ported to rust version https://github.com/Blockstream/gdk/blob/gdk-electrum/subprojects/gdk_rust/gdk_common/src/wally/mod.rs#L504)

stevenroose commented:
I added the test. I also added `Address::to_confidential` and `Address::to_unconfidential`. I hope they can just be added in this PR without need for a separate one.

RCasatta commented:
ack f0bc17c
